ext/http/lib.rs

```diff
@@ -186,10 +186,29 @@ impl HttpConnResource {
           .unwrap_or(Encoding::Identity)
       };
 
+      let content_length = if request
+        .headers()
+        .get(hyper::header::CONTENT_ENCODING)
+        .is_none()
+      {
+        request
+          .headers()
+          .get(hyper::header::CONTENT_LENGTH)
+          .and_then(|v| v.to_str().ok())
+          .and_then(|v| v.parse::<u64>().ok())
+      } else {
+        None
+      };
```

Comment:
This can be replaced with a call to [`Request::size_hint()`](https://docs.rs/hyper/latest/hyper/body/trait.HttpBody.html#method.size_hint). This will return the lower and upper bound for the body size. You can return this upper and lower bound from `HttpStreamResource::size_hint()`.

Reply:
Done @lucacasonato, PTAL